zerver/tests/test_email_change.py

```diff
@@ -41,13 +46,13 @@ def test_confirm_email_change_when_time_exceeded(self) -> None:
                                                old_email=old_email,
                                                user_profile=user_profile,
                                                realm=user_profile.realm)
-        key = generate_key()
+
         date_sent = now() - datetime.timedelta(days=2)
-        Confirmation.objects.create(content_object=obj,
-                                    date_sent=date_sent,
-                                    confirmation_key=key,
-                                    type=Confirmation.EMAIL_CHANGE)
-        url = confirmation_url(key, user_profile.realm, Confirmation.EMAIL_CHANGE)
+        url = create_confirmation_link(obj, Confirmation.EMAIL_CHANGE)
+        registration_key = url.split('/')[-1]
+        Confirmation.objects.filter(
+            confirmation_key=registration_key).update(date_sent=date_sent)
```

Comment:
What's the story with this code change?  Might be better as a preparatory commit since it seems partially to be a cleanup to improve the test.